msrestazure/azure_operation.py

```diff
@@ -154,23 +169,26 @@ def _is_empty(self, response):
     def _deserialize(self, response):
         """Attempt to deserialize resource from response.
 
-        :raises: OperationFailed if deserialized resource has status of
-         failed or cancelled.
-        :raises: OperationFinished if deserialised resource has status
-         succeeded.
+        :param requests.Response response: latest REST call response.
         """
-        self.resource = self.get_outputs(response)
-        if self.method == 'PUT':
-            resource_status = self._get_resource_status()
-            if failed(resource_status):
-                self.status = resource_status
-                raise OperationFailed("Operation failed or cancelled")
-            elif succeeded(resource_status):
-                raise OperationFinished("Operation succeeded")
-            elif resource_status:
-                self.status = resource_status
-
-    def _get_body_status(self, response):
+        # Hacking response with initial status_code
+        previous_status = response.status_code
+        response.status_code = self.initial_status_code
+        resource = self.get_outputs(response)
+        response.status_code = previous_status
+
+        # Hack for Storage mostly, to workaround the bug in the Python generator
+        if resource is None:
+            try:
+                previous_status = response.status_code
+                response.status_code = 200
+                resource = self.get_outputs(response)
+                response.status_code = previous_status
+            except Exception:
```

Comment:
We can catch the mother of all msrest exception (ClientRuntime error I think). Might want to catch JSONDecodeError, but I think JSON validity has been tested before.

FYI this will be fixed by https://github.com/Azure/msrestazure-for-python/issues/11